File: blockchain-analytics-flink/src/main/java/io/github/anttikaikkonen/blockchainanalyticsflink/casssandra/models/Transaction.java
package io.github.anttikaikkonen.blockchainanalyticsflink.casssandra.models;

import com.datastax.driver.mapping.annotations.Column;
import com.datastax.driver.mapping.annotations.PartitionKey;
import com.datastax.driver.mapping.annotations.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Table(name="transaction")
public class Transaction {
    @PartitionKey
    String txid;
    
    int height;
    
    @Column(name="tx_n")
    int txN;
    
    int size;
    long version;
    long locktime;
    
    int input_count;
    int output_count;
    long tx_fee;
}

File: blockchain-graphql-server/src/models/transaction.ts
import { ObjectType, Field, Int } from 'type-graphql'
import { Coin } from './coin'

@ObjectType()
export class Transaction {

    @Field({ nullable: false, complexity: 1 })
    txid: string

    @Field(type => Int, { nullable: true, complexity: 1 })
    height: number

    @Field(type => Int, { nullable: true, complexity: 1 })
    txN: number

    @Field(type => Int, { nullable: false, complexity: 1 })
    size: number

    @Field(type => Int, { nullable: false, complexity: 1 })
    version: number

    @Field(type => Int, { nullable: false, complexity: 1 })
    lockTime: number

    @Field({ nullable: false, complexity: 1 })
    fee: number

    @Field(type => Int, { nullable: false, complexity: 1 })
    inputCount: number

    @Field(type => Int, { nullable: false, complexity: 1 })
    outputCount: number

    @Field({ nullable: false, complexity: 1 })
    readonly coin: Coin

}


